contracts/eosio.system/include/eosio.system/peer_keys.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <eosio/contract.hpp>
+#include <eosio/crypto.hpp>
+#include <eosio/name.hpp>
+
+#include <string>
+#include <optional>
+
+namespace eosiosystem {
+
+using eosio::name;
+using eosio::public_key;
+
+// -------------------------------------------------------------------------------------------------
+// -------------------------------------------------------------------------------------------------
+struct [[eosio::table("peerkeys"), eosio::contract("eosio.system")]] peer_key {
+   struct v0_data {
+      std::optional<public_key> pubkey; // peer key for network message authentication
+      EOSLIB_SERIALIZE(v0_data, (pubkey))
+   };
+
+   name                  account;
+   std::variant<v0_data> data;
+
+   uint64_t primary_key() const { return account.value; }
+
+   void                                    set_public_key(const public_key& key) { data = v0_data{key}; }
+   const std::optional<eosio::public_key>& get_public_key() const {
+      return std::visit([](auto& v) -> const std::optional<eosio::public_key>& { return v.pubkey; }, data);
+   }
+   void update_row() {}
+   void init_row(name n) { *this = peer_key{n, v0_data{}}; }
```

Comment:
Don't really see the value of either of these methods. Why does `init_row` not take a `public_key` ?

Reply:
These are meant to support a future where we would add other optional data to the table. The various actions like `regpeerkey` would not have to change, just the `update_row()` and `init_row()` methods. The `update_row()` was more useful when it updated `modif_block_time`,  but it could concievably still be useful.